robots/ComauSmartSixRobot.m

```diff
@@ -32,13 +32,15 @@
             comau_DH_d =     [-0.45, 0, 0, -0.64707, 0, -0.095];
             comau_DH_a =     [0, 0.150, 0.590, 0.13, 0, 0];
             comau_DH_alpha = [pi, pi/2, pi, -pi/2, -pi/2, pi/2];
+            comau_DH_type =  [1,1,1,1,1,1];
```

Comment:
Please create a `JointType` [enumeration class](https://uk.mathworks.com/help/matlab/matlab_oop/enumerations.html) containing all joints described in Silva, Quiroz-Omaña, and Adorno (2022). Then use something like JointType.revolute instead of just `1`.

Reply:
@dqrobotics/developers 

Hi @bvadorno I created the class as you suggested. The class has the following format:

```matlab
classdef JointType < double

   % This class is based on Table 1 of  Silva, Quiroz-Omaña, and Adorno (2022).
   % Dynamics of Mobile Manipulators Using Dual Quaternion Algebra
   % https://doi.org/10.1115/1.4054320
   enumeration
      REVOLUTE    (1)
      PRISMATIC   (2)
      SPHERICAL   (3)
      CYLINDRICAL (4)
      PLANAR      (5)
      SIX_DOF     (6)
      HELICAL     (7)
   end
end

```

Usage:

```matlab
JointType.REVOLUTE 
```

Alternative:

If we want to have something like:

```matlab
DQ_SerialManipulatorDH.JointType.REVOLUTE; % matlab,
```

```cpp
DQ_SerialManipulatorDH::JointType::REVOLUTE; // C++,
```

we could have a struct property. For instance, 

```matlab
properties (Constant)
     JointType = struct('REVOLUTE',1, 'PRISMATIC',2, ... , 'HELICAL', 7);   
end
```

What do you think ?

Reply:
Hi @juanjqo,

I think an enumeration class is more appropriate because joints can be used in robots other than manipulators. Also, the advantage of using an enumeration class is that we can then define methods and other properties that hold for particular joints.

I suggest inheriting from `uint32` instead of `double`. Also, to maintain consistency with our terminology, let's call it `DQ_JointType`. I had first suggested using `JointType` because this is an auxiliary enumeration class. But now I imagine this class having interesting properties and methods (for defining friction and backlash, etc.)

Best wishes,
Bruno

Reply:
@bvadorno I implemented your suggestions.